pkg/apis/policy/v1alpha1/clusterimagepolicy_validation.go

```diff
@@ -278,11 +278,23 @@ func (p *Policy) Validate(ctx context.Context) *apis.FieldError {
 	if p.Type != "cue" && p.Type != "rego" {
 		errs = errs.Also(apis.ErrInvalidValue(p.Type, "type", "only [cue,rego] are supported at the moment"))
 	}
-	if p.Data == "" && p.ConfigMapRef == nil {
-		errs = errs.Also(apis.ErrMissingField("data", "configMapRef"))
+	if p.Data == "" && p.ConfigMapRef == nil && p.URL == nil {
+		errs = errs.Also(apis.ErrMissingField("data", "configMapRef", "url"))
 	}
-	if p.Data != "" && p.ConfigMapRef != nil {
-		errs = errs.Also(apis.ErrMultipleOneOf("data", "configMapRef"))
+	if p.Data != "" && p.ConfigMapRef != nil && p.URL != nil {
+		errs = errs.Also(apis.ErrMultipleOneOf("data", "configMapRef", "url"))
+	}
+	if (p.Data != "" && p.ConfigMapRef != nil) ||
+		(p.Data != "" && p.URL != nil) ||
+		(p.ConfigMapRef != nil && p.URL != nil) {
+		errs = errs.Also(apis.ErrMultipleOneOf("data", "configMapRef", "url"))
+	}
+	if p.URL != nil {
+		url := *p.URL
+		_, err := apis.ParseURL(url.String())
```

Comment:
I can't recall if this allows for things like relative URLs? Also, wonder if this should be required to be https?

Reply:
I can easily improve the validation here. Yes, i like the https requirement here

Reply:
I added some additional validations.